node/src/main/scala/org/bitcoins/node/SpvNode.scala

```diff
@@ -0,0 +1,77 @@
+package org.bitcoins.node
+
+import akka.actor.ActorSystem
+import org.bitcoins.chain.api.ChainApi
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+import org.bitcoins.db.AppConfig
+import org.bitcoins.node.models.Peer
+import org.bitcoins.node.networking.Client
+import org.bitcoins.node.networking.peer.{
+  PeerMessageReceiver,
+  PeerMessageSender
+}
+import org.bitcoins.rpc.util.AsyncUtil
+
+import scala.concurrent.Future
+
+case class SpvNode(peer: Peer, chainApi: ChainApi)(
+    implicit appConfig: AppConfig,
+    system: ActorSystem) {
+  import system.dispatcher
+
+  private val peerMsgRecv = PeerMessageReceiver.newReceiver(appConfig)
+
+  private val client: Client =
+    Client(context = system, peer = peer, peerMessageReceiver = peerMsgRecv)
+
+  private val peerMsgSender: PeerMessageSender = {
+    PeerMessageSender(client, appConfig.network)
+  }
+
+  /** Starts our spv node */
+  def start(): Future[SpvNode] = {
+    peerMsgSender.connect()
```

Comment:
I'm not really familiar with Akka actors. But this immediately seems a little fishy. Is there no way this can go wrong?

Reply:
akka implies the "send and forget" message passing style. This means that we do not know what happens after we call `client.actor ! Tcp.Connect(socket)` 

You need to look at [PeerMessageReceiver.isConnected](https://github.com/bitcoin-s/bitcoin-s/blob/node/node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageReceiver.scala#L94) to see if the connection was successful.

Reply:
Although looking at this closer, it does look like we just looking to see if the [underlying promise](https://github.com/bitcoin-s/bitcoin-s/blob/node/node/src/main/scala/org/bitcoins/node/networking/peer/PeerMessageReceiverState.scala#L38) is completed, not what it is actually completed with. This probably should be changed in the future

Reply:
Issue filed here to address this in the future https://github.com/bitcoin-s/bitcoin-s/issues/482